### ISSUE TITLE
RetinaNet experience with localization of text

### ISSUE THREAD

**Issue by ghost** [fizyr/keras-retinanet#796]
Hello and thank you for your efforts to make things simple.
I have trained icdar2015 data through your repo
You left everything default
I already got a good test accuracy
The implementation time is very standard

- Here is my trial report
-Training data 1000 images 
-test data 500 images
About 11,000 reliable positions and 17,000 false positions

The results are
from IOU [icdar2015](rrc.cvc.uab.es/?ch=4&com=evaluation&task=15) ?
`Calculated!{"recall": 0.5406836783822822, "precision": 0.6864303178484108, "hmean": 0.6049016967411797, "AP": 0} resnet50 , Epoch = 15 ,score = 0.2`
,,
`{"recall": 0.4997592681752528, "precision": 0.7483777937995674, "hmean": 0.5993071593533487, "AP": 0} resnet50 , Epoch = 50 ,score = 0.2`
''
```
{"recall": 0.4232065479056331, "precision": 0.802007299270073, "hmean": 0.554049795146549, "AP": 0} resnet50 , Epoch = 15 ,score = 0.8
{"recall": 0.18921521425132404, "precision": 0.9401913875598086, "hmean": 0.31503006012024054, "AP": 0} resnet50 , Epoch = 15 ,score = 0.9
{"recall": 0.08329321136254213, "precision": 0.9719101123595506, "hmean": 0.15343680709534366, "AP": 0} resnet50 , Epoch = 15 ,score = 1.0
```

and from retinanet _evaluation_ mAP
resnet50  Epoch 50 
`mAP: 0.4359`

Time on images? :
```
 [500 /500] ~:100.0%
 processing time:  167.96209335327148
```
Time on Video? :
```
elasped time: 291.42
approx FPS  4.98
```
RetinaNet vs [East](https://github.com/argman/EAST)
```East fps:
elasped time: 608.22
approx FPS  2.39
​ RetinaNet fps:
elasped time: 291.42
approx FPS  4.98
```
=======
But after reviewing the results manually I noticed a very excellent performance for Retinanet
The first problem I have found so far is dealing with very small texts
The second problem is the rotation text
The form can find the position of the text but can not be determined correctly.

First I love to share my experience with you:
Secondly, I hope that these results are better some day.
I know it may be difficult but I hope you will give me the starting line for the route I'm taking
I aspire to the following (find the round text and improve the detection of small texts)
'images result'

![img_136 1](https://user-images.githubusercontent.com/44137873/48517172-28999300-e86e-11e8-8578-78a3865b87bd.jpg)
![img_10 14](https://user-images.githubusercontent.com/44137873/48517176-29322980-e86e-11e8-840c-0cc67a114e84.jpg)
![oop 5](https://user-images.githubusercontent.com/44137873/48517177-29cac000-e86e-11e8-9dd3-81293836a544.jpg)
![e8cc521a-ff03-4d87-8648-d363f6f19c7c](https://user-images.githubusercontent.com/44137873/48517568-63e89180-e86f-11e8-9402-528c7d559a2b.jpg)
![d9930afb-60b8-414b-b992-30fd2624fab7](https://user-images.githubusercontent.com/44137873/48517573-66e38200-e86f-11e8-823f-3cecfade89ab.jpg)
![8d331111-b0cf-4e80-8da4-d083f86bc173](https://user-images.githubusercontent.com/44137873/48517574-6945dc00-e86f-11e8-8829-6248b0463c2a.jpg)

thank you

**Comment by hgaiser** [comment]
Hey! Thanks for sharing the results and happy to hear you're getting some nice results. You could try to improve your results by adding `--random-transform` to enable data augmentation. It will randomly rotate your images, so it might help in dealing with rotated text as well. Good luck!

**Comment by mohammed-Emad** [comment]
@hgaiser Thank you so much ,I will try this
